src/details/ArborX_DetailsTags.hpp

```diff
@@ -45,6 +45,12 @@ struct Tag<Box>
   using type = BoxTag;
 };
 
+template <typename T>
+struct Tag<T>
```

Comment:
Also are their other types of tags you would expect? The name seems pretty generic. This is OK unless it would be possible for a type to have more than one `Tag` depending on how it is used.